python_modules/dagster/dagster/_core/storage/upath_io_manager.py

```diff
@@ -251,9 +251,13 @@ def _load_single_input(
                         f" {backcompat_path}"
                     )
                 except FileNotFoundError:
-                    raise e
+                    # if none of the paths exist, then we assume the handle_output was skipped. This would
+                    # happen if the output was None, so we provide a None.
+                    return None
```

Comment:
went with this approach (returning None in the try/except) rather than checking for file existence first for 2 reasons:
1. the try/except was already in this function
2. some of the implementations (azure/aws/...) use mock resources for testing and the `path.exists()` calls would fail when using mock resources. Fixing it would have maybe required using real buckets, which felt unnecessary when we could just put the `return None` somewhere else